packages/es-dev-server/src/middleware/plugin-transform.ts

```diff
@@ -53,5 +104,15 @@ export function createPluginTransformMiddlware(cfg: PluginServeMiddlewareConfig)
         }
       }
     }
+
+    if (cfg.fileWatcher && transformCache && cfg.cache && cfg.cacheKeysForFilePaths) {
```

Comment:
If these are not optional parameters, we don't need to check if they're there.

Reply:
I removed the cache and cacheKeysForFilePaths check, I think this was added as a lack of typescript experience, I had some type errors because I defined them as `<thing> | undefined`